streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -565,11 +566,13 @@ public void process(final ProcessorSupplier<? super K, ? super V> processorSuppl
         KStreamImpl<K, V1> joinOther = (KStreamImpl<K, V1>) other;
 
         if (joinThis.repartitionRequired) {
-            joinThis = joinThis.repartitionForJoin(joined.keySerde(), joined.valueSerde());
+            final String leftJoinRepartitionTopicName = joined.name() != null ? joined.name() + "-left" : null;
```

Comment:
Could we set `name` instead of `null` here and simplify other code (cf. my comments below) ?

Reply:
ack